aten/src/ATen/native/Resize.cpp

```diff
@@ -21,15 +22,29 @@ Tensor& resize_cpu_(Tensor& self, IntArrayRef size) {
 // this function in native_functions.yaml
 Tensor& resize_as_sparse_(Tensor& self, const Tensor& src);
 
-Tensor& resize_as_(Tensor& self, const Tensor& the_template) {
+Tensor& resize_as_(
+    Tensor& self,
+    const Tensor& the_template,
+    c10::optional<MemoryFormat> optional_memory_format) {
+  auto memory_format =
+      optional_memory_format.value_or(MemoryFormat::Contiguous);
   if (self.is_sparse() && the_template.is_sparse()) {
+    TORCH_CHECK(
+        memory_format == MemoryFormat::Contiguous,
+        "Unsupported memory format for sparse tensor resize_as_ ",
+        memory_format);
     return native::resize_as_sparse_(self, the_template);
   }
   Tensor& result = self.resize_(the_template.sizes());
+  if (memory_format == MemoryFormat::Preserve) {
```

Comment:
documentation?

It's not obvious to me how memory_format even applies to resize_, given the crazy semantics of resize.  Can you document all the behaviors (sharing X non_contiguous X larger/smaller) and what this means?

I suspect we are only trying to support some subcases of this.

Reply:
I will put documentation, but generally speaking I had to convert this function as some models and developers is using it similar to empty_like, but resizing existing tensor. So to make this models BC compatible we need to preserve memory format similarly.

Reply:
I get that but I'm not convinced we want the specific implementation here.

Reply:
Updated in the PR description